ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,935 @@
+):
+    return aggcontext.agg(data, type(op).__name__.lower())
+
+
+variance_ddof = {'pop': 0, 'sample': 1}
+
+
+@execute_node.register(ops.Variance, SeriesGroupBy, type(None))
+def execute_reduction_series_groupby_var(
+    op, data, _, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(data, 'var', ddof=variance_ddof[op.how])
+
+
+@execute_node.register(ops.StandardDev, SeriesGroupBy, type(None))
+def execute_reduction_series_groupby_std(
+    op, data, _, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(data, 'std', ddof=variance_ddof[op.how])
+
+
+@execute_node.register(
+    (ops.CountDistinct, ops.HLLCardinality), SeriesGroupBy, type(None)
+)
+def execute_count_distinct_series_groupby(
+    op, data, _, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(data, 'nunique')
+
+
+@execute_node.register(ops.Arbitrary, SeriesGroupBy, type(None))
+def execute_arbitrary_series_groupby(op, data, _, aggcontext=None, **kwargs):
+    how = op.how
+    if how is None:
+        how = 'first'
+
+    if how not in {'first', 'last'}:
+        raise com.OperationNotDefinedError(
+            'Arbitrary {!r} is not supported'.format(how)
+        )
+    return aggcontext.agg(data, how)
+
+
+def _filtered_reduction(mask, method, data):
+    return method(data[mask[data.index]])
+
+
+@execute_node.register(ops.Reduction, SeriesGroupBy, SeriesGroupBy)
+def execute_reduction_series_gb_mask(
+    op, data, mask, aggcontext=None, **kwargs
+):
+    method = operator.methodcaller(type(op).__name__.lower())
+    return aggcontext.agg(
+        data, functools.partial(_filtered_reduction, mask.obj, method)
+    )
+
+
+@execute_node.register(
+    (ops.CountDistinct, ops.HLLCardinality), SeriesGroupBy, SeriesGroupBy
+)
+def execute_count_distinct_series_groupby_mask(
+    op, data, mask, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(
+        data,
+        functools.partial(_filtered_reduction, mask.obj, dd.Series.nunique),
+    )
+
+
+@execute_node.register(ops.Variance, SeriesGroupBy, SeriesGroupBy)
+def execute_var_series_groupby_mask(op, data, mask, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data,
+        lambda x, mask=mask.obj, ddof=variance_ddof[op.how]: (
+            x[mask[x.index]].var(ddof=ddof)
+        ),
+    )
+
+
+@execute_node.register(ops.StandardDev, SeriesGroupBy, SeriesGroupBy)
+def execute_std_series_groupby_mask(op, data, mask, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data,
+        lambda x, mask=mask.obj, ddof=variance_ddof[op.how]: (
+            x[mask[x.index]].std(ddof=ddof)
+        ),
+    )
+
+
+@execute_node.register(ops.Count, DataFrameGroupBy, type(None))
+def execute_count_frame_groupby(op, data, _, **kwargs):
+    result = data.size()
+    # FIXME(phillipc): We should not hard code this column name
+    result.name = 'count'
+    return result
+
+
+@execute_node.register(ops.Reduction, dd.Series, (dd.Series, type(None)))
```

Comment:
would for sure group by op here (as hard to tell what is defiend for ops.Reduction for example)